pkg/psmdb/mongos_test.go

```diff
@@ -0,0 +1,310 @@
+package psmdb
+
+import (
+	"context"
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	api "github.com/percona/percona-server-mongodb-operator/pkg/apis/psmdb/v1"
+	"github.com/percona/percona-server-mongodb-operator/version"
+)
+
+func TestMongosHost(t *testing.T) {
+	ctx := context.Background()
+
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "mongos-0",
+			Namespace: "default",
+		},
+	}
+
+	tests := map[string]struct {
+		init          func(cl client.Client)
+		cr            *api.PerconaServerMongoDB
+		expectedHost  string
+		expectedError error
+	}{
+		"service not found": {
+			init: func(cl client.Client) {},
+			cr: &api.PerconaServerMongoDB{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-cluster",
+					Namespace: "default",
+				},
+				Spec: api.PerconaServerMongoDBSpec{
+					ClusterServiceDNSSuffix: "svc.cluster.local",
+					Sharding: api.Sharding{
+						Mongos: &api.MongosSpec{
+							Expose: api.MongosExpose{
+								ServicePerPod: false,
+							},
+						},
+					},
+				},
+			},
```

Comment:
This is also not affecting the tests since it remains contant, we can extract it to 

```
			cr := &api.PerconaServerMongoDB{
				ObjectMeta: metav1.ObjectMeta{
					Name:      "test-cluster",
					Namespace: "default",
				},
				Spec: api.PerconaServerMongoDBSpec{
					ClusterServiceDNSSuffix: "svc.cluster.local",
					Sharding: api.Sharding{
						Mongos: &api.MongosSpec{
							Expose: api.MongosExpose{
								ServicePerPod: false,
							},
						},
					},
				},
```

Reply:
personally prefer adding it inside the test Run(...) because it keeps each test isolated in terms of data and if we want to add something in the future there configurable for a specific test, it will be easy.